Source/DafnyLanguageServer/README.md

```diff
@@ -75,3 +75,18 @@ Options provided through the command line have higher priority than the options
 # Default: true (mark the statements)
 --ghost:markStatements=true
 ```
+
+### Plugins
```

Comment:
Could you say something about what you can do with plugins here? I think it might be good for users to know that they can use this to implement custom diagnostics, but not custom refactorings, for example. Maybe something like what is described here: https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/compiler-api-model#diagnostic-apis

Reply:
> custom refactorings or tree-to-tree transformations

With a refactoring I meant enabling extra "Code actions" to be shown in the IDE that trigger a rewrite of source code.

Is there a use-case for doing a tree-to-tree transformation before resolution finishes, if it doesn't enable adding new syntax? If we're not sure then I'd prefer hiding those methods. We can make them as `internal` instead of `public` to hide them.

Reply:
I am not sure about the use case. I prefer to remove the extra API for now until it is requested to us. I splitted Plugin rewriters from internal rewriters now, so that we can be cherry-picking about the API to expose.